internal/resource/registry.go

```diff
@@ -0,0 +1,71 @@
+package resource
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+type Registry interface {
+	// Register the given resource type and its hooks.
+	Register(reg Registration)
+
+	// Resolve the given resource type and its hooks.
+	Resolve(typ *pbresource.Type) (reg Registration, ok bool)
+}
```

Comment:
Ah, guessing you needed to include these files because they're not present in my branch? Sorry!